src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -137,6 +137,26 @@ private RecommendedSymbols GetSymbolsOffOfContainer()
                 }
             }
 
+            if ((node.GetAncestor<IsPatternExpressionSyntax>() is { } isPattern && isPattern.Pattern.Span.Contains(node.Span)) ||
+                node.CheckParent<BinaryExpressionSyntax>(s => s is BinaryExpressionSyntax(SyntaxKind.IsExpression) && s.Right.Span.Contains(node.Span)))
```

Comment:
instead of checking span containment (no clue what taht actually means here), just check how the target token relates to the expression you're looking at.

Reply:
We want to make sure that we are on the right hand side of the `is` pattern, so we validate that there. `IsPatternExpressionSyntax` could be more complex, so in order to avoid traversal and enumeration, we use the span check to quickly evaluate that.

Reply:
dont' look up arbitrarily far with GetAncestor.  We should just check for the exact constructs we expect.

Reply:
> I think there should be button to fix that IntelliSense and remove the Error from the Error List.

we should be explicit on what we are trying to match against.  This allows for finding something anywhere up, regardless of how high it is, which is not ok.

for example, you could have something like `x is .... { ... [ ... [... $$here$$ ... ] } ] }`.  This would count in your world for the first check.  But that doesn't at all imply that the next check and logic is ok.

Reply:
> dont' look up arbitrarily far with GetAncestor. We should just check for the exact constructs we expect.

How can we know how far to look up for? We could be deep down into the RHS of the constant pattern.

> This would count in your world for the first check. But that doesn't at all imply that the next check and logic is ok.

So how else would we determine from the discovered ancestor that we are in the RHS of the expression?

Reply:
> We could be deep down into the RHS of the constant pattern.

You can write an algorithm to walk upwards where appropriate. That algorithm can contain a loop of need. That algorithm cannot contain arbitrary code that looks upwards an unbounded amount, while not checking what sorts of nodes are there along the way.

Reply:
Adjusted, ptal